Helper/PaymentMethods.php

```diff
@@ -398,38 +307,115 @@ protected function getCurrentShopperReference()
     }
 
     /**
-     * @return array|mixed
-     * @throws \Adyen\AdyenException
+     * @param $merchantAccount
+     * @param \Magento\Store\Model\Store $store
+     * @param $country
+     * @param \Magento\Quote\Model\Quote $quote
+     * @return array
+     * @throws \Exception
      */
-    public function getConnectedTerminals()
-    {
-        $storeId = $this->session->getQuote()->getStoreId();
+    protected function getPaymentMethodsRequest(
+        $merchantAccount,
+        \Magento\Store\Model\Store $store,
+        $country,
+        \Magento\Quote\Model\Quote $quote
+    ) {
+        $currencyCode = $this->chargedCurrency->getQuoteAmountCurrency($quote);
 
-        // initialize the adyen client
-        $client = $this->adyenHelper->initializeAdyenClient($storeId, $this->adyenHelper->getPosApiKey($storeId));
+        $paymentMethodRequest = [
+            "channel" => "Web",
+            "merchantAccount" => $merchantAccount,
+            "countryCode" => $this->getCurrentCountryCode($store, $country),
+            "shopperLocale" => $this->adyenHelper->getCurrentLocaleCode($store->getId()),
+            "amount" => [
+                "currency" => $currencyCode
+            ]
+        ];
 
-        // initialize service
-        $service = $this->adyenHelper->createAdyenPosPaymentService($client);
+        if (!empty($this->getCurrentShopperReference())) {
+            $paymentMethodRequest["shopperReference"] = $this->getCurrentShopperReference();
+        }
 
-        $requestParams = [
-            "merchantAccount" => $this->adyenHelper->getAdyenMerchantAccount('adyen_pos_cloud', $storeId),
-        ];
+        $amountValue = $this->adyenHelper->formatAmount($this->getCurrentPaymentAmount(), $currencyCode);
```

Comment:
In this brach we already use the latest library where the currency sanitize function is available to get the minor units for an amount.

The adyenHelper->formatAmount can be replaced with the function from the library and the helper functions can be removed from the adyenHelper like: https://github.com/Adyen/adyen-magento2/blob/7.0.0-rc.1/Helper/Data.php#L279-L312 and https://github.com/Adyen/adyen-magento2/blob/7.0.0-rc.1/Helper/Data.php#L321-L324

Reply:
Ticket created PW-3512